netbox/circuits/urls.py

```diff
@@ -2,48 +2,49 @@
 
 from dcim.views import CableCreateView, CableTraceView
 from extras.views import ObjectChangeLogView
+from utilities.urls import cached
 from . import views
 from .models import Circuit, CircuitTermination, CircuitType, Provider
 
 app_name = 'circuits'
 urlpatterns = [
 
     # Providers
-    url(r'^providers/$', views.ProviderListView.as_view(), name='provider_list'),
+    url(r'^providers/$', cached(views.ProviderListView.as_view()), name='provider_list'),
```

Comment:
It's probably cleanest to keep caching functions out of `urls.py`. We can wrap `get()` inside ObjectListView to knock out all the object lists, and wrap `get()` for each of the individual object views.